src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -549,4 +550,28 @@
    (let [handler-maybe-with-authorization (if use-legacy-auth-conf
                                             route-handler
                                             (authorization-fn route-handler))]
-     (wrap-middleware handler-maybe-with-authorization puppet-version))))
\ No newline at end of file
+     (wrap-middleware handler-maybe-with-authorization puppet-version))))
+
+(schema/defn ^:always-validate
+  construct-root-routes-handler :- bidi-schema/RoutePair
```

Comment:
just a route tree. i think calling it `construct-root-routes` makes sense to me.